blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/nav-section.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { NAV_BREAKPOINTS, PLACEMENT_AREAS } from '../nav-helper';
+import WidgetList from './widget-list';
+
+const NavSection = ({
+  children,
+  customFields,
+  getNavWidgetType,
+  hasUserConfiguredNavItems,
+  menuButtonClickAction,
+  side,
+  signInOrder,
+}) => (
+  // istanbul ignore next
+  !side ? null : (
+    <div key={side} className={`nav-${side}`}>
+      {
+        // Support for deprecated 'signInOrder' custom field
+        // "If" condition is for rendering "signIn" element
+        // "Else" condition is for standard nav bar customization logic
+        side === 'right'
+        && !hasUserConfiguredNavItems(getNavWidgetType)
+        && signInOrder
+        && Number.isInteger(signInOrder)
+        && children[signInOrder - 1]
+          ? children[signInOrder - 1]
+          : NAV_BREAKPOINTS.map((breakpoint) => (
+            <div key={breakpoint} className={`nav-components--${breakpoint}`}>
+              <WidgetList
+                breakpoint={breakpoint}
+                customFields={customFields}
+                getNavWidgetType={getNavWidgetType}
+                id={side}
+                menuButtonClickAction={menuButtonClickAction}
+                placement={PLACEMENT_AREAS.NAV_BAR}
+              >
+                {children}
```

Comment:
importantly, we're passing in props this way rather than relying upon children as a global variable